modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -103,11 +138,43 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
             .asScala
             .zipWithIndex
             .map { case (value, index) =>
-              EnumValue(value.getValue(), index, value.getName().asScala)
+              EnumValue(
+                value.getValue(),
+                index,
+                EnumUtil.enumValueClassName(
+                  value.getName().asScala,
+                  value.getValue,
+                  index
+                )
+              )
             }
             .toList
           Enumeration(shape.name, shape.name, values).some
-        case _ => super.stringShape(shape)
+        case _ => getDefault(shape)
+      }
+
+      override def enumShape(shape: EnumShape): Option[Decl] = {
+        val values = shape
+          .getEnumValues()
+          .asScala
+          .zipWithIndex
+          .map { case ((name, value), index) =>
+            EnumValue(value, index, name)
+          }
+          .toList
+        Enumeration(shape.name, shape.name, values).some
+      }
+
+      override def intEnumShape(shape: IntEnumShape): Option[Decl] = {
+        val values = shape
+          .getEnumValues()
+          .asScala
+          .zipWithIndex
```

Comment:
in the case of `IntEnum`, the value should become the ordinal, and I guess the `Name` should take the place of the string value.

Reply:
Okay will make this change. Just to clarify, this will mean that the ordinal will be used to send over the wire whenever the IntEnum hint is present, correct?

Reply:
I guess the ordinal will no longer be guaranteed to be an actual ordinal, meaning that the ordinal values may not be sequential and may not start at 0.. seems like that breaks the principle of least surprise, wdyt?